test_cephfs_sr.py

```diff
@@ -0,0 +1,119 @@
+            assert False, "SR creation should not have succeeded!"
+        except:
+            print("SR creation failed, as expected.")
+
+    # Impact on other tests: installs cephfsprogs and creates the SR
+    def test_create_sr(self, host_with_cephfsprogs):
+        host = host_with_cephfsprogs
+        TestCephFSSR.sr = host.sr_create('cephfs', "CephFS-SR", self.device_config)
+        wait_for(TestCephFSSR.sr.exists, "Wait for SR to exist")
+
+    # Impact on other tests: creates a VM on the SR and starts it
+    def test_import_and_start_VM(self, host):
+        vm = host.import_vm_url(vm_image('mini-linux-x86_64-bios'), TestCephFSSR.sr.uuid)
+        TestCephFSSR.vm = vm # for teardown
+        #vm.convert_type('pv')
+        vm.start()
+        vm.wait_for_os_booted()
+
+    # Impact on other tests: none if succeeds
+    # FIXME: only suited to linux VMs
+    def test_snapshot(self, host):
+        vm = TestCephFSSR.vm
+        vm.test_snapshot_on_running_linux_vm()
+
+    # Impact on other tests: VM shutdown cleanly
+    def test_vm_shutdown(self, host):
+        vm = TestCephFSSR.vm
+        vm.shutdown(verify=True)
+
+    # *** tests with reboots (longer tests). To be moved to another file?
+
+    # Impact on other tests: none if succeeds
+    def test_reboot(self, host):
+        host.reboot(verify=True)
+        wait_for(TestCephFSSR.sr.all_pbds_attached, "Wait for PBD attached")
+        vm = TestCephFSSR.vm
+        vm.start()
+        vm.wait_for_os_booted()
+        vm.shutdown(verify=True)
+
+    # Impact on other tests: none if succeeds
+    def test_cephfsprogs_missing(self, host):
+        sr = TestCephFSSR.sr
+        cephfsprogs_installed = True
+        try:
+            host.yum_remove(['ceph-common', 'centos-release-ceph-jewel'])
+            cephfsprogs_installed = False
+            try:
+                sr.scan()
+                assert False, "SR scan should have failed"
+            except:
+                print("SR scan failed as expected.")
+            host.reboot(verify=True)
+            # give the host some time to try to attach the SR
+            time.sleep(10)
+            print("Assert PBD not attached")
+            assert not sr.all_pbds_attached()
+            host.yum_install(['centos-release-ceph-jewel'])
+            host.yum_install(['ceph-common'])
+            cephfsprogs_installed = True
+            sr.plug_pbds(verify=True)
+            sr.scan()
+        finally:
+            if not cephfsprogs_installed:
+                host.yum_install(['centos-release-ceph-jewel'])
+                host.yum_install(['ceph-common'])
+
+    # *** End of tests with reboots
+
+    # Impact on other tests: VM removed, leaving SR empty (and thus destroyable)
+    def test_destroy_vm(self, host):
+        if TestCephFSSR.vm is not None:
+            TestCephFSSR.vm.destroy(verify=True)
+
+    # Impact on other tests: SR destroyed
+    # Prerequisites: SR attached but empty
+    def destroy_sr(self, host):
```

Comment:
bug fixed in the latest version of test_xfs_sr.py: should be named `test_destroy_sr`